syntax = "proto2";

package library;

message InventoryItem {
  optional string id = 1;

  oneof inventory_type {
    Book book = 2;
  }

  enum Status {
    AVAILABLE = 0;
    TAKEN = 1;
  }

  optional Status status = 3;
}

message Book {
  optional string ISBN = 1;
  optional string title = 2;
  optional string author = 3;
  repeated string genre = 4;
  optional int32 year = 5;
}

service InventoryService {
  rpc CreateBook(BookCreateRequest) returns (BookCreateResponse) {}
  rpc GetBook(BookGetRequest) returns (BookGetResponse) {}
}

message BookCreateRequest {
  optional string id = 1;
  optional Book book = 2; 
  optional InventoryItem.Status status = 3;
}

message BookCreateResponse {
  optional string message = 1;
}

message BookGetRequest {
  optional string ISBN = 1;
}

message BookGetResponse {
  optional string message = 1;
}
